machine/v1beta1/types_awsprovider.go

```diff
@@ -107,6 +107,22 @@ type AWSMachineProviderConfig struct {
 	// If this value is selected, capacityReservationID must be specified to identify the target reservation.
 	// +optional
 	MarketType MarketType `json:"marketType,omitempty"`
+
+	// hostID specifies the Dedicated Host on which the instance must be started.
+	// This field is mutually exclusive with DynamicHostAllocation.
+	// +kubebuilder:validation:Pattern=`^h-[0-9a-f]{17}$`
+	// +kubebuilder:validation:MaxLength=19
+	// +openshift:enable:FeatureGate=AWSDedicatedHosts
+	// +optional
+	HostID *string `json:"hostID,omitempty"`
+
+	// hostAffinity specifies the dedicated host affinity setting for the instance.
+	// When HostAffinity is set to host, an instance started onto a specific host always restarts on the same host if stopped.
+	// When HostAffinity is set to default, and you stop and restart the instance, it can be restarted on any available host.
+	// When HostAffinity is defined, HostID is required.
```

Comment:
Whether it is set to `host` or `default`, `hostID` will be required?

Also, when referring to a field use the serialized form of the field name as that is what end-users would be familiar with:

```suggestion
	// When HostAffinity is defined, hostID is required.
```

Reply:
To clarify, `hostID` is required when `hostAffinity` is set to `host` and should be forbidden otherwise?